src/main/java/seedu/address/logic/commands/transaction/EditTransactionCommand.java

```diff
@@ -0,0 +1,259 @@
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("index", index)
+                .add("editTransactionDescriptor", editTransactionDescriptor)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the transaction with. Each non-empty field value will replace the
+     * corresponding field value of the transaction.
+     */
+    public static class EditTransactionDescriptor {
+        private Name name;
+        private Amount amount;
+        private Category category;
+        private DateTime datetime;
+        private Location location;
+        private Type type;
+
+        public EditTransactionDescriptor() {}
+
+        /**
+         * Copy constructor.
+         */
+        public EditTransactionDescriptor(EditTransactionCommand.EditTransactionDescriptor toCopy) {
+            setName(toCopy.name);
+            setAmount(toCopy.amount);
+            setCategory(toCopy.category);
+            setDateTime(toCopy.datetime);
+            setLocation(toCopy.location);
+            setType(toCopy.type);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(name, amount, category, datetime, location, type);
+        }
+
+        public void setName(Name name) {
+            this.name = name;
+        }
+
+        public Optional<Name> getName() {
+            return Optional.ofNullable(name);
+        }
+
+        public void setAmount(Amount amount) {
+            this.amount = amount;
+        }
+
+        public Optional<Amount> getAmount() {
+            return Optional.ofNullable(amount);
+        }
+
+        public void setCategory(Category category) {
+            this.category = category;
+        }
+
+        public Optional<Category> getCategory() {
+            return Optional.ofNullable(category);
+        }
+
+        public void setDateTime(DateTime datetime) {
+            this.datetime = datetime;
+        }
+
+        public Optional<DateTime> getDateTime() {
+            return Optional.ofNullable(datetime);
+        }
+
+        public void setLocation(Location location) {
+            this.location = location;
+        }
+
+        public Optional<Location> getLocation() {
+            return Optional.ofNullable(location);
+        }
+
+        public void setType(Type type) {
+            this.type = type;
+        }
+
+        public Optional<Type> getType() {
+            return Optional.ofNullable(type);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof EditTransactionCommand.EditTransactionDescriptor)) {
+                return false;
```

Comment:
same as above, can we add a simple unit test? maybe smth like this:

    EditTransactionCommand.EditTransactionDescriptor descriptor = 
            new EditTransactionCommand.EditTransactionDescriptor();

        assertTrue(...);